src/Agent.Plugins/PipelineArtifact/FileShareProvider.cs

```diff
@@ -103,21 +161,33 @@ private async Task CopyFileShareAsync(string downloadRootPath, string destPath,
                         cancellationToken: cancellationToken);
 
                 executionContext.Output(StringUtil.Loc("RobocopyBasedPublishArtifactTaskExitCode", exitCode));
-
                 // Exit code returned from robocopy. For more info https://blogs.technet.microsoft.com/deploymentguys/2008/06/16/robocopy-exit-codes/
                 if (exitCode >= 8)
                 {
                     throw new Exception(StringUtil.Loc("RobocopyBasedPublishArtifactTaskFailed", exitCode));
                 }
+
+                return new FileSharePublishResult (robocopyArguments, exitCode);
             }
         }
 
-        private async Task DownloadFileShareArtifactAsync(string sourcePath, string destPath, int parallelCount, CancellationToken cancellationToken, IEnumerable<Func<string, bool>> minimatchFuncs = null)
+        private async Task<ArtifactRecord> DownloadFileShareArtifactAsync(
+            string sourcePath,
+            string destPath,
+            int parallelCount,
+            CancellationToken cancellationToken,
+            IEnumerable<string> minimatchPatterns = null)
         {
+            Stopwatch watch = Stopwatch.StartNew();
+
+            IEnumerable<Func<string, bool>> minimatchFuncs = MinimatchHelper.GetMinimatchFuncs(minimatchPatterns, this.tracer);
+
             var trimChars = new[] { '\\', '/' };
 
             sourcePath = sourcePath.TrimEnd(trimChars);
 
+            var artifactName =  new DirectoryInfo(destPath).Name;
+
```

Comment:
Have artifactName passed by the caller instead of directly taking it out of the destination path because this breaks separation of concerns/decentralizes where the artifact name logic is determined.